astropy/version.py

```diff
@@ -11,7 +11,14 @@
     )
     del warnings
 
-    version = "0.0.0"
+    # this branch may be executed in rare but still valid conditions. For instance,
+    # if astropy is installed from a shallow clone, i.e., without most of the vcs
+    # history, which is needed to compute the dev version number dynamically.
+
+    # default to a virtually infinite version number, never to be reached, that should
+    # - alert the reader if seen in production
+    # - satisfy any real-world lower bound on astropy's version
+    version = "999.999.999"
```

Comment:
Errr... this is unusual default. Can't we just set it to empty string? If this route is reached, things should break anyway.

Reply:
the empty string wouldn't work either. What things would you say *should* break in that case ?

Reply:
In practice, I have a private repo that has astropy as a submodule, and in that context I never could figure out how to unshallow the submodule, so I always get the default string, and then need to workaround anything broken by it (which has never been relevant as far as I'm aware).

Reply:
I know it's unusual, and I hesitated to propose this patch for a while, but I must have hit at least 3 of 4 very niche problems this would have avoided, ,so at some point it becomes a question of professional integrity: if I don't propose it, I accept that I'll keep wasting time and money over problems I could have avoided.

So, practically, what are the important properties this default version string should have, that I didn't already list ?
Is there any reason to assume anyone would *rely* on the current behavior ?

Reply:
(and in case it helps understanding where I'm coming from: I don't "use" astropy as a submodule, rather, this is my way to implement a "private fork" that requires little to no maintenance to keep in sync, and where I can iterate quickly on CI innovations as experiments that I then upstream)